lib/hyrax/transactions/steps/save.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+require 'dry/monads'
+
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # Saves a given work, returning a Result (Success|Failure)
+      class Save
+        include Dry::Monads[:result]
+
+        ##
+        # @params [#save] persister
+        def initialize(persister: Hyrax.persister)
+          @persister = persister
+        end
+
+        ##
+        # @param [Hyrax::ChangeSet] change_set
+        #
+        # @return [Dry::Monads::Result] `Success(work)` if the change_set is
+        #   applied and the resource is saved;
+        #   `Failure([#to_s, change_set.resource])`, otherwise.
+        def call(change_set)
+          saved = @persister.save(resource: change_set.sync)
+
+          depositor = ::User.find_by_user_key(saved.try(:depositor))
```

Comment:
This looks a bit odd…I'm reading this as "depositor" is optional.

Reply:
that's right. i'm still fighting a bit with this flow.

The event payload wants to have the user responsible for the change (or `nil`, if none (?)), but the transaction doesn't already have it.

While Hyrax Core Metadata demands a depositor, it's strongly preferable not to raise an error after saving in this step, regardless of the input. The object has been saved, and we *really* want to publish the appropriate events whatever else may happen. Since the transaction can't guarantee that the `ChangeSet` validation will support depositor, it seems necessary to be resilient to its absence and to the method missing case.

Reply:
@no-reply Could you add that to the code comments? I think the context would be extremely helpful for what sounded like working lore of the Core Metadata implementation

Reply:
what do you think about the `with_step_args` option mentioned in the PR comment?

i'm thinking it's very likely we'll completely rip up this line, so i'm not inclined to spend much time refining it or commenting on it. but maybe the short-term solution is to just always pass `nil` as the user until we have a way for the caller to pass that in?

Reply:
Reading your current PR post and in light of this comment, the PR and your comment make sense. Good work. I do want @cjcolvar to weigh in as he's often mentioned Dry::Transaction and other actor stack related work that he's done; He's done work closer to this PRs area of concern.